src/sdc11073/pysoap/soapenvelope.py

```diff
@@ -1,50 +1,40 @@
-from io import BytesIO
-from typing import Optional, List
+from __future__ import annotations
+
 from enum import Enum
+from typing import TYPE_CHECKING
+
 from lxml import etree as etree_
 
-from ..exceptions import ApiUsageError
-from ..namespaces import default_ns_helper as ns_hlp
-from ..xml_types import xml_structure as struct
-from ..xml_types.basetypes import XMLTypeBase, MessageType, ElementWithText
+from sdc11073.exceptions import ApiUsageError
+from sdc11073.namespaces import default_ns_helper as ns_hlp
+from sdc11073.xml_types import xml_structure as struct
+from sdc11073.xml_types.basetypes import ElementWithText, MessageType, XMLTypeBase
+
+if TYPE_CHECKING:
+    from sdc11073.pysoap.msgreader import PayloadData
+    from sdc11073.xml_types.addressing_types import HeaderInformationBlock
 
 CHECK_NAMESPACES = False  # can be used to enable additional checks for too many namespaces or undefined namespaces
 
 
-class SoapResponseException(Exception):
+class SoapResponseError(Exception):
+    """Exception raised when Response could not be processed."""
 
-    def __init__(self, response_envelope):
+    def __init__(self, response_envelope: ReceivedSoapMessage | PayloadData):
         super().__init__()
         self.response_envelope = response_envelope
 
 
 class ExtendedDocumentInvalid(etree_.DocumentInvalid):
-    pass
-
-
-_LANGUAGE_ATTR = '{http://www.w3.org/XML/1998/namespace}lang'
-
-
-def _assert_valid_exception_wrapper(schema, content):
-    try:
-        schema.assertValid(content)
-    except etree_.DocumentInvalid:
-        # reformat again to produce better error output
-        tmp_str = etree_.tostring(content, pretty_print=True)
-        tmp = etree_.parse(BytesIO(tmp_str))
-        tmp_str = tmp_str.decode('utf-8')
-        try:
-            schema.assertValid(tmp)
-        except etree_.DocumentInvalid as err:
-            msg = f'{str(err)}\n{tmp_str}'
-            raise ExtendedDocumentInvalid(msg, error_log=err.error_log) from err
+    """Exception for invalid document."""
 
 
 class Soap12Envelope:
-    """This represents an outgoing soap envelope"""
+    """Soap12Envelope represents an outgoing soap envelope."""
```

Comment:
why only outgoing

Reply:
there is a different type for received messages.